### PR TITLE
Added list of standard tokens owned by user

### PR DESCRIPTION
**Pull request by DavidGershony**

### CODE REVIEW

File: angular/src/app/account/account.component.ts [hunk 1]
```diff
@@ -116,10 +122,10 @@ export class AccountComponent implements OnInit, OnDestroy {
     // Reset the account state when performing a re-scan.
     accountState.balance = 0;
     this.accountStateStore.set(account.identifier, accountState);
-    this.accountStateStore.save();
+    await this.accountStateStore.save();
```
Comment (sondreb) [inline angular/src/app/account/account.component.ts:125]: Good catch!

File: angular/src/app/services/wallet-manager.ts [hunk 2]
```diff
@@ -1,36 +1,37 @@
 // import { HDKey } from "micro-bip32"; // TODO: Uninstall the previous package, replaced with @scure.
-import { HDKey } from '@scure/bip32';
-import { mnemonicToSeedSync } from '@scure/bip39';
+import {HDKey} from '@scure/bip32';
```
Comment (sondreb) [inline angular/src/app/services/wallet-manager.ts:2]: Here your editor is clearly not formatting like VS Code =)

File: angular/src/app/services/wallet-manager.ts [hunk 3]
```diff
@@ -797,7 +800,22 @@ export class WalletManager {
...
+    if (network.smartContractSupport) {
+      await this.LoadStandardTokensForAccountAsync(network, this.getAddressByIndex(account, 0, 0));
+    }
+  }
+
+  async LoadStandardTokensForAccountAsync(network: Network, address: string) {
+    const indexerUrl = this.networkLoader.getServer(network.id, this.settings.values.server, this.settings.values.indexer);
+
+    const tokens = await axios.get(`${indexerUrl}/api/query/${network.name}/tokens/${address}`);
+    if (tokens.data.items) {
```
Comment (sondreb) [inline angular/src/app/services/wallet-manager.ts:815]: What if previously you received data, but no more, shouldn't it be removed then?

**Review by sondreb — APPROVED** [review]
Few minor comments

File: angular/src/app/services/wallet-manager.ts (hunk 3 above)
Comment (DavidGershony) [reply angular/src/app/services/wallet-manager.ts:815]: What do you mean? should I empty the store before updating it?

File: angular/src/app/services/wallet-manager.ts (hunk 2 above)
Comment (DavidGershony) [reply angular/src/app/services/wallet-manager.ts:2]: My bad.. will look into it

File: angular/src/app/services/wallet-manager.ts (hunk 3 above)
Comment (sondreb) [reply angular/src/app/services/wallet-manager.ts:815]: If I have tokens, and I sell them, won't it still tell me that I have tokens that I no longer have? If items is null, 0, undefined, it won't be updated in the store. So if I had a token yesterday, sold it today, will it disappear or not?

File: angular/src/app/services/wallet-manager.ts [hunk 4]
```diff
@@ -797,7 +800,22 @@ export class WalletManager {
...
+      for (let token of tokens.data.items)
+        this.tokensStore.set(token.name, token);
```
Comment (sondreb) [inline angular/src/app/services/wallet-manager.ts:817]: Does this distinguish between wallets and accounts? Are all tokens stored globally and accessible for all wallets? Should be filtered based upon accounts.

File: angular/src/app/services/wallet-manager.ts (hunk 3 above)
Comment (DavidGershony) [reply angular/src/app/services/wallet-manager.ts:815]: I changed the flow to clear the tokens when fetching new ones for the account

File: angular/src/app/services/wallet-manager.ts (hunk 4 above)
Comment (DavidGershony) [reply angular/src/app/services/wallet-manager.ts]: Changed the tokens to be stored with the account identifier as key holding the list of tokens under it

File: angular/src/app/services/wallet-manager.ts [hunk 5]
```diff
@@ -797,7 +800,24 @@ export class WalletManager {
...
+    }
+  }
+
+  async LoadStandardTokensForAccountAsync(network: Network, account: Account) {
+    const indexerUrl = this.networkLoader.getServer(network.id, this.settings.values.server, this.settings.values.indexer);
+    const address = this.getReceiveAddressByIndex(account, 0);
+
+    const tokens = await axios.get(`${indexerUrl}/api/query/${network.name}/tokens/${address.address}`);
+    if (tokens.data.items) {
+      this.tokensStore.remove(account.identifier);
```
Comment (sondreb) [inline angular/src/app/services/wallet-manager.ts:817]: This is not needed, set should overwrite the existing one =)

**Review by sondreb — APPROVED** [review]
Great work! Let's get it merged!

**Pull request merged**